src/input/input_python.rs

```diff
@@ -643,45 +642,14 @@ impl<'a> Input<'a> for PyAny {
     }
 }
 
-/// return None if obj is not a mapping (cast_as::<PyMapping> fails or mapping.items returns an AttributeError)
+/// return None if obj is not a mapping (cast_as::<PyMapping> fails)
 /// otherwise try to covert the mapping to a dict and return an Some(error) if it fails
 fn mapping_as_dict(obj: &PyAny) -> Option<ValResult<GenericMapping>> {
-    let mapping: &PyMapping = match obj.cast_as() {
+    let mapping = match obj.cast_as() {
```

Comment:
Maybe I should change the function name like `cast_as_mapping` or?

Reply:
sounds good.